File: packages/cli/test/fixtures/ranking-corpora/python/rankingpy/handlers/create.py
"""Validates and records a new user by name."""

from rankingpy.util.logger import log_error, log_info


def create_user(name: str) -> None:
    if not name:
        log_error("create_user", ValueError("name must not be empty"))
        return
    log_info(f"created user {name}")

File: packages/cli/test/fixtures/ranking-corpora/rust/src/create.rs
use crate::util;

/// Validates and records a new user by name, logging the outcome through the
/// shared `util` module.
pub fn create_user(name: &str) -> Result<(), String> {
    if name.is_empty() {
        let err = "name must not be empty".to_string();
        util::log_error("create_user", &err);
        return Err(err);
    }
    util::log_info(&format!("created user {}", name));
    Ok(())
}
